[tools] Replace clang_format.py with clang_format_new.py

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34701 [JIT] clang-format JIT code
* **#34700 [tools] Replace clang_format.py with clang_format_new.py**

Summary:
This commit replaces clang_format.py with clang_format_new.py, a
new and improved script that downloads, verifies and runs a platform-appropriate
clang-format binary on files in a predefined set of whitelisted directories.

Testing:
Ran the script.

Commit:
[tools] Replace clang_format.py with clang_format_new.py

Summary:
This commit replaces clang_format.py with clang_format_new.py, a
new and improved script that downloads, verifies and runs a platform-appropriate
clang-format binary on files in a predefined set of whitelisted directories.

Testing:
Ran the script.

[ghstack-poisoned]

suo reviewed (approved):
pre-emptively accepting

Commit:
Update on "[tools] Replace clang_format.py with clang_format_new.py"


Summary:
This commit replaces clang_format.py with clang_format_new.py, a
new and improved script that downloads, verifies and runs a platform-appropriate
clang-format binary on files in a predefined set of whitelisted directories.

Testing:
Ran the script.

[ghstack-poisoned]
